src/task.c

```diff
@@ -168,6 +215,17 @@ typedef struct {
    platform_heap_id heap_id;
 } thread_invoke;
 
+/*
+ * -----------------------------------------------------------------------------
+ * task_invoke_with_hooks() -
+ *
+ * A thread has been created with a call-back function to execute. This
+ * function invokes that function bracketted by calls to register / deregister
+ * the thread with Splinter's task system.
+ *
+ * RESOLVE: I see references to this, ... but who calls this?
```

Comment:
Let's discuss .... this, and then there seem to be a few more functions, that are written but never called.

Is this vestigial code that we could / should get rid of now, to reduce confusion ... [???].